### PR TITLE
fix: make compile query public

### PR DESCRIPTION
**Pull request by owlas**

### CODE REVIEW

File: packages/backend/package.json [hunk 1]
```diff
@@ -71,8 +71,6 @@
         "cron-converter": "^2.0.0",
         "cronstrue": "^2.23.0",
         "csv-stringify": "^6.2.4",
-        "duckdb": "^0.9.2",
```
Comment (owlas) [inline packages/backend/package.json:74]: no longer needed

File: packages/backend/src/inMemoryTableCalculations.ts [hunk 2]
```diff
@@ -1,134 +0,0 @@
-import { DimensionType } from '@lightdash/common';
```
Comment (owlas) [inline packages/backend/src/inMemoryTableCalculations.ts:1]: no longer needed

File: packages/backend/src/queryBuilder.test.ts [hunk 3]
```diff
@@ -64,6 +64,7 @@ import {
     METRIC_QUERY_WITH_TABLE_REFERENCE,
     METRIC_QUERY_WITH_TABLE_REFERENCE_SQL,
     MONTH_NAME_SORT_SQL,
+    QUERY_BUILDER_UTC_TIMEZONE,
```
Comment (owlas) [inline packages/backend/src/queryBuilder.test.ts:67]: signature change

File: packages/backend/src/queryBuilder.ts [hunk 4]
```diff
@@ -650,12 +642,22 @@ export type CompiledQuery = {
     fields: ItemsMap;
 };
 
+export type BuildQueryProps = {
```
Comment (owlas) [inline packages/backend/src/queryBuilder.ts:645]: move closer to implementation

File: packages/backend/src/queryBuilder.ts [hunk 5]
```diff
@@ -650,12 +642,22 @@ export type CompiledQuery = {
...
+    explore: Explore;
+    compiledMetricQuery: CompiledMetricQuery;
+    warehouseClient: WarehouseClient;
+    userAttributes?: UserAttributeValueMap;
+    intrinsicUserAttributes: IntrinsicUserAttributes;
+    timezone: string;
```
Comment (owlas) [inline packages/backend/src/queryBuilder.ts:651]: needs a timezone for query

File: packages/backend/src/queryBuilder.ts [hunk 6]
```diff
@@ -668,7 +670,6 @@ export const buildQuery = ({
         limit,
         additionalMetrics,
         compiledCustomDimensions,
-        timezone,
```
Comment (owlas) [inline packages/backend/src/queryBuilder.ts:671]: don't use `metricQuery.timezone`

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 7]
```diff
@@ -916,154 +904,25 @@ export class ProjectService extends BaseService {
...
-    }
-
-    private async _compileQuery(
+    static async _compileQuery(
         metricQuery: MetricQuery,
         explore: Explore,
         warehouseClient: WarehouseClient,
         intrinsicUserAttributes: IntrinsicUserAttributes,
         userAttributes: UserAttributeValueMap,
+        timezone: string,
```
Comment (owlas) [inline packages/backend/src/services/ProjectService/ProjectService.ts:913]: - requires `timezone` 
- still `static`
- not `private`

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 8]
```diff
@@ -1796,44 +1600,16 @@ export class ProjectService extends BaseService {
                         warehouseClient.runQuery(
                             query,
                             queryTags,
-                            metricQuery.timezone,
+                            // metricQuery.timezone,
```
Comment (owlas) [inline packages/backend/src/services/ProjectService/ProjectService.ts:1603]: disable warehouse timezone change behaviour

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 9]
```diff
@@ -1887,24 +1663,6 @@ export class ProjectService extends BaseService {
...
-                     * If the feature flag is enabled, and we actually have any table calculations
-                     * to process, we use the new in-memory table calculations engine. This avoid us
-                     * spinning-up a new DuckDB database pointlessly.
-                     *
-                     * In a follow-up after initial testing, this check should be done somewhere further
-                     * down the stack.
-                     */
-                    const newTableCalculationsFeatureFlagEnabled =
-                        await isFeatureFlagEnabled(
-                            FeatureFlags.UseInMemoryTableCalculations,
```
Comment (ZeRego) [inline packages/backend/src/services/ProjectService/ProjectService.ts:1900]: We can remove the feature flag `UseInMemoryTableCalculations` from `FeatureFlags`

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 10]
```diff
@@ -1955,73 +1713,19 @@ export class ProjectService extends BaseService {
...
-                    );
-                    const timezoneMetricQuery = {
-                        ...metricQueryWithLimit,
-                        timezone: isTimezoneEnabled
-                            ? metricQueryWithLimit.timezone
-                            : undefined,
-                    };
-                    const compileQueryArgs = [
-                        timezoneMetricQuery,
+                    const { timezone } = this.lightdashConfig.query;
```
Comment (ZeRego) [inline packages/backend/src/services/ProjectService/ProjectService.ts:1716]: (not a blocker) Is this used?

**Review by ZeRego — APPROVED** [review]

File: packages/backend/src/services/ProjectService/ProjectService.ts (hunk 10 above)
Comment (owlas) [reply packages/backend/src/services/ProjectService/ProjectService.ts:1716]: nope! removing

**Pull request merged**